include/inviwo/core/datastructures/image/layerram.h

```diff
@@ -271,11 +280,24 @@ template <typename T>
 LayerRAMPrecision<T>::LayerRAMPrecision(const LayerRAMPrecision<T>& rhs)
     : LayerRAM(rhs)
     , dimensions_(rhs.dimensions_)
-    , data_(new T[dimensions_.x * dimensions_.y])
+    , data_(std::make_unique<T[]>(glm::compMul(dimensions_)))
     , swizzleMask_(rhs.swizzleMask_)
     , interpolation_{rhs.interpolation_}
     , wrapping_{rhs.wrapping_} {
-    std::memcpy(data_.get(), rhs.data_.get(), dimensions_.x * dimensions_.y * sizeof(T));
+    std::copy(rhs.getView().begin(), rhs.getView().end(), data_.get());
```

Comment:
could use `std::ranges::copy`
```suggestion
    std::ranges::copy(rhs.getView(), data_.get());
```